x-pack/test/detection_engine_api_integration/security_and_spaces/bundled_prebuilt_rules_package/install_latest_bundled_prebuilt_rules.ts

```diff
@@ -0,0 +1,83 @@
+import path from 'path';
+
+import expect from 'expect';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import {
+  deleteAllPrebuiltRuleAssets,
+  deleteAllRules,
+  getPrebuiltRulesAndTimelinesStatus,
+} from '../../utils';
+import { BUNDLED_PACKAGE_DIR } from './config';
+import { downloadPackageFromFleet } from '../../utils/prebuilt_rules/download_prebuilt_rules_from_epr';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext): void => {
+  const es = getService('es');
+  const supertest = getService('supertest');
+  const log = getService('log');
+
+  const removeBundledPackages = async (bundledFleetPackageDir: string) => {
+    const files = await fs.readdir(bundledFleetPackageDir);
+
+    // Do not remove mock packages with version 99.0.0 and 99.0.1
+    // which are used to test the installation of prerelease packages
+    const downloadedFiles = files.filter((file) => !file.includes('99')) || [];
+
+    for (const file of downloadedFiles) {
+      await fs.unlink(path.join(bundledFleetPackageDir, file));
+    }
+  };
+
+  /* This test simulates an air-gapped environment in which the user doesn't have access to EPR.
+  /* We first download the package from the registry as done during build time, and then
+  /* attempt to install it from the local file system. The API response from EPM provides
+  /* us with the information of whether the package was installed from the registry or
+  /* from a package that was bundled with Kibana */
+  describe('install_bundled_prebuilt_rules', () => {
+    beforeEach(async () => {
+      await deleteAllRules(supertest, log);
+      await deleteAllPrebuiltRuleAssets(es);
+    });
+
+    afterEach(async () => {
+      await removeBundledPackages(BUNDLED_PACKAGE_DIR);
+    });
+
+    it('should install prebuilt rules from the package that comes bundled with Kibana', async () => {
+      // Verify that status is empty before package installation
+      const statusBeforePackageInstallation = await getPrebuiltRulesAndTimelinesStatus(supertest);
+      expect(statusBeforePackageInstallation.rules_installed).toBe(0);
+      expect(statusBeforePackageInstallation.rules_not_installed).toBe(0);
+      expect(statusBeforePackageInstallation.rules_not_updated).toBe(0);
+
+      // Download the package from the registry as done during build time,
+      // according to the fleet packages configured in fleet_packages.json file
+      const { version } = await downloadPackageFromFleet(BUNDLED_PACKAGE_DIR, log);
+
+      const EPM_URL = `/api/fleet/epm/packages/security_detection_engine/${version}`;
+
+      const bundledInstallResponse = await supertest
+        .post(EPM_URL)
+        .set('kbn-xsrf', 'xxxx')
+        .type('application/json')
+        .send({ force: true })
+        .expect(200);
+
+      // As opposed to "registry"
+      expect(bundledInstallResponse.body._meta.install_source).toBe('bundled');
```

Comment:
This is what I mentioned in a previous comment, that we actually want to test that we successfully can install the assets from a bundled package. I agree that the whole logic of reading the `fleet_packages.json` file and then downloading from Fleet is too intrincate and probably shouldn't be part of our test suite.

But still I think that the installation from bundle is an important part of the test plan. How would this test look for you?  It would simply read the `fleet_packages.json` file, and make an assertion that `security_detection_engine` is there?

Reply:
The alternative is what the Fleet team does: uploading a package in a zip and forcing Fleet to install that. Then we can make assertions that the package has `install_source` of `bundled`.
This is what we are already doing for testing the large package and rule upgrades.